### PR TITLE
Add Realm.Configuration.cache

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Expose Realm Core's `cache` configuration option.

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: docs/realm.js [hunk 1]
```diff
@@ -426,6 +426,7 @@ class Realm {
  *    what fits in memory, but it is not persistent and will be removed when the last instance
  *    is closed. This option is incompatible with `sync`.
  * @property {boolean} [readOnly=false] - Specifies if this Realm should be opened as read-only.
+ * @property {boolean} [cache=true] - Cache Realm instances.
```
Comment (bmunkholm) [inline docs/realm.js:429]: Couldn't this use an explanation as to when to cache and when not to?

Comment (bmunkholm) [reply docs/realm.js:429]: Were we caching previously?

Comment (tgoyne) [reply docs/realm.js:429]: No, this is something we specifically do not expose publicly.

**Review by takameyer — APPROVED** [review]
LGTM

File: docs/realm.js (hunk 1 above)
Comment (takameyer) [reply docs/realm.js:429]: There's a lint error here that should be fixed.

Comment (nirinchev) [reply docs/realm.js:429]: I think you should fix/reference count close rather than expose cache.

Comment (takameyer) [reply docs/realm.js:429]: It's a workaround until we have the time and resources to fix said issue.

Comment (bmunkholm) [reply docs/realm.js:429]: Is the plan then to make a breaking change later to remove the option once this real issue is fixed?

As a minimum, there has to be a descriptive parameter description.

**Pull request closed without merging**